### PR TITLE
Throw NotAuthorizedException if a JWT is sent to an unauthenticated endpoint and is not valid

### PR DESCRIPTION
**Pull request by radcortez**
To fix TCK test EmptyTokenTest in MP 1.2.

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java [hunk 1]
```diff
@@ -62,6 +61,7 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
 
             if (bearerToken != null) {
                 try {
+                    requestContext.setProperty(HAS_JWT, true);
```
Comment (sberyozkin) [inline implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java:64]: I've looked at this filter, it catches the exception, logs, and continues, it needs to return 401 immediately, so a new property, and a new optional filter won't be needed

Comment (radcortez) [reply implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java:64]: The idea was to follow the same structure we had with the others filters. Sure, I can change it.

Comment (sberyozkin) [reply implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java:64]: Yeah, if we have a token and it failed to parse then relying on some follow up filters to block such an invalid token will not be very safe...

Comment (sberyozkin) [reply implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java:64]: @radcortez In quarkus we may need to relax things a bit, here we have a proactive authentication, but in Quarkus Stuart added an option to authenticate lazily, but we can discuss it later in Quarkus

Comment (radcortez) [reply implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java:64]: Sure. Quarkus doesn't use this filter anyway.

**Pull request merged**